#170 missing success msg bug fix

ratulkai93 opened the pull request:
The current branch will display a success message after a successful file upload into the database. 

This also fixes a small bug for devs:
After uploading a dataset, only "U" would be logged in the console browser because of Line 28 -  "console.log(result[0])" in Client/src/Components/DataCell/DataCell.tsx . With 19fdace , the whole message is logged with "console.log(result)". The message is "Upload was successful!" (Line 203) from Server/src/services/fileUploadService.ts

ratulkai93 committed:
[#170] added alert snackbar for successfull file upload

will display green pop up bar for uploading any json file currently

ratulkai93 committed:
Merge branch 'develop' into #170-missing-success-msg-bug-fix

ratulkai93 committed:
[#170] added fix from develop for front-back comm

ratulkai93 committed:
[#170] fixed console message for successful upload

ratulkai93 committed:
Merge branch 'develop' into #170-missing-success-msg-bug-fix

ratulkai93 committed:
[#170] fixed console log message bug

currently shows message: Upload was successful!

GD-Dheer commented on Client/src/Components/DataCell/DataCell.js:
```diff
@@ -29,6 +30,7 @@ export default function DataCell() {
             setOpen(true)
             return
         }
+        setAlertSuccess(true)
```
Are we setting the alert if the user is able to input a file into the upload box or if the upload was successful?

ratulkai93 replied:
we are setting it to true if the file extension check passes (ie. file.json), so it will open the snackbar and display the success message on the page, then turn it to false.

GD-Dheer commented on Client/src/Components/DataCell/DataCell.js:
```diff
@@ -42,7 +44,7 @@ export default function DataCell() {
                 .then(resp => resp.json())
                 .then(result => {
-                    console.log(result[0])
+                    console.log(result)
```
No need to log here

JeremiahTee replied:
I asked him to change this because the logged message was incomplete. In the console browser, after uploading a dataset, it would log "U" only instead of "Upload was successful!"

ratulkai93 replied:
I don't think its a good idea to remove the console logs at the moment since they are mostly for troubleshooting the file upload and im assuming other people are working with it as well. 
If required, i'll remove them on a future commit.

GD-Dheer commented on Client/src/Components/DataCell/DataCell.js:
```diff
@@ -66,9 +68,15 @@ export default function DataCell() {
 
+           <Snackbar open={alertSuccess} autoHideDuration={3000} onClose={() => setAlertSuccess(false)}>
+                <Alert onClose={() => setAlertSuccess(false)} severity="success">
+                    File Successfully updated!! ^_^
```
A+

GD-Dheer commented on Client/src/Components/DataCell/DataCell.tsx:
```diff
@@ -25,7 +25,7 @@ export default function DataCell() {
         .then(resp => resp.json())
         .then(result => {
-          console.log(result[0])
+          console.log(result)
```
Another log

GD-Dheer commented on Client/src/Components/DataCell/FileUploadForm.tsx:
```diff
@@ -31,6 +33,8 @@ export const FileUploadForm = (props: IProps) => {
     console.log("vF is " + validateFile + "  and vF() is " + validateFile(file))
     if (isValidFile) {
       onSubmit(file)
+      console.log('File successfully uploaded')
```
Log..

GD-Dheer commented on Server/src/routes/fileUploadRouter.ts:
```diff
@@ -16,7 +16,7 @@ let router = Router();
  * the service.
  */
-router.post('/dataupload', [JWTAuthenticator.verifyJWT, upload.single('file')], (request: MulterRequest, response: Response, next: NextFunction) => {
+router.post('/dataupload', upload.single('file'), (request: MulterRequest, response: Response, next: NextFunction) => {
```
This change cant be commited - has to keep JWTAuthenticator.verifyJWT...

ratulkai93 replied:
Had to remove that line before because the file upload would not work when its there. its now re added on the recent commit.

GD-Dheer reviewed (approved):
Nice fix/. Thx

Mehrdadmaskull reviewed (approved):
Looks good, only comment is bring back the JWT auth

ratulkai93 committed:
[#170] Re-added JWT auth inside file upload router

this line was previously removed for testing purposes. should address the PR comment and fix the block.

ratulkai93 committed:
[#170] minor fix

modified file upload success msg

ratulkai93 committed:
Merge branch 'develop' into #170-missing-success-msg-bug-fix
